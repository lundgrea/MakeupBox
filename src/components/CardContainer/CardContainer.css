.card-container{
  display: flex;
}

.results-category-headline{
  text-align: center;
}

.results-category-box {
  border: 2px solid black;
  border-radius: 2px;
  padding: 10px;
  margin: 10px;
  background: black;
  color: white;
}