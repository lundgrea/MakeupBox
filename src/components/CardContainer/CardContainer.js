import React from 'react';
import './CardContainer.css';
import Cards from '../Cards/Cards';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { handleErrors } from '../../actions'


export const CardContainer = ({error, categories, data}) => {

  const buildCardDeck = (data) => {
    if (!data) {
      return error
    } else {
      return data.map((cardInfo, index)=> {
        return (
          <div key={index}>
            <Cards 
            {...cardInfo}
            key={cardInfo.id}
              />
          </div>
        )
      })
    }
  }
    
  return (
    <>
    <article className='card-container'>
      <section key='0' className='results-category-box'>
        <h2 className='results-category-headline'>{categories[0]}</h2>
        {buildCardDeck(data[0])}
      </section>
      <section key='1' className='results-category-box'>
        <h2 className='results-category-headline'>{categories[1]}</h2>
        {buildCardDeck(data[1])}
      </section>
      <section key='2' className='results-category-box'>
        <h2 className='results-category-headline'>{categories[2]}</h2>
        {buildCardDeck(data[2])}
      </section>
    </article>
    </>
  );
};

export const mapStateToProps = store => ({
  error: store.hasErrored
})

export const mapDispatchToProps = dispatch => ({
  handleErrors: error => dispatch(handleErrors(error))
});

export default connect(mapStateToProps, mapDispatchToProps)(CardContainer);

CardContainer.propTypes = {
  data:PropTypes.array,
  categories: PropTypes.array 
};