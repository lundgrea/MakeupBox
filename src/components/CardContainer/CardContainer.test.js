import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer } from './CardContainer';

describe('CardContainer', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<CardContainer 
        category="lips"
        data={[ [{},{},{},{},{},{}], [{},{},{},{},{},{}], [{},{},{},{},{},{}] ]}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});