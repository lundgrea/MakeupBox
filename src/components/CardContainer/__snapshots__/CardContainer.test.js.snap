// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all data passed in correctly 1`] = `
<article
  className="card-container"
>
  <section
    className="results-category-box"
    key="0"
  >
    <h2
      className="results-category-headline"
    >
      l
    </h2>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
  </section>
  <section
    className="results-category-box"
    key="1"
  >
    <h2
      className="results-category-headline"
    >
      i
    </h2>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
  </section>
  <section
    className="results-category-box"
    key="2"
  >
    <h2
      className="results-category-headline"
    >
      p
    </h2>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
    <div>
      <Cards />
    </div>
  </section>
</article>
`;
